src/main/scala/com/microsoft/hyperspace/index/sources/interfaces.scala

```diff
@@ -57,8 +58,11 @@ trait FileBasedRelation extends SourceRelation {
    *
    * This API is used when the signature of source needs to be computed, e.g., creating an index,
    * computing query plan's signature, etc.
+   *
+   * If it is not possible to compute the signature (e.g. there are no files left),
+   * the implementation might return None.
    */
-  def signature: String
+  def signature(fb: FingerprintBuilder): Option[Fingerprint]
```

Comment:
Please add the new parameter in the scaladoc with `@param`.

Reply:
Done